frontend/vre/utils/jsonld.model.test.js

```diff
@@ -0,0 +1,81 @@
+import assert from 'assert';
+import sinon from 'sinon';
+import {nestSubject} from "./jsonld.model";
+
+function findById(graph, id) {
+    return graph.find((subject) => subject["@id"] === id);
+}
+
+const exampleJsonLDGraph = [{
+    "@id": "http://example.com/s1",
+    "dc:title": "Title without references",
+}, {
+    "@id": "http://example.com/s2",
+    "dc:title": "Title with one internal reference",
+    "dc:description": {
+        "@id": "http://example.com/descForS2",
+    },
+}, {
+    "@id": "http://example.com/s3",
+    "dc:title": "Title with one external (or missing) reference",
+    "dc:description": {
+        "@id": "http://example.com/somethingExternal",
+    },
+}, {
+    "@id": "http://example.com/s4",
+    "dc:title": "Title with one nested internal reference",
+    "dc:description": {
+        "@id": "http://example.com/descForS4",
+    },
+}, {
+    "@id": "http://example.com/s5",
+    "dc:title": "Title with sameAs s6",
+    "owl:sameAs": {
+        "@id": "http://example.com/s6",
+    },
+}, {
+    "@id": "http://example.com/s6",
+    "dc:title": "Title with sameAs s5",
+    "owl:sameAs": {
+        "@id": "http://example.com/s5",
+    },
+}, {
+    "@id": "http://example.com/descForS4",
+    "example:value": {
+        "@id": "http://example.com/descForDescForS4"
+    }
+}, {
+    "@id": "http://example.com/descForDescForS4",
+    "example:value": "Random description",
+}, {
+    "@id": "http://example.com/descForS2",
+    "example:value": "Random description",
+}];
+
+describe('nestSubject', () => {
+    it('does not change anything if there are no references', () => {
+        const subject = findById(exampleJsonLDGraph, "http://example.com/s1");
+        const ns = nestSubject(exampleJsonLDGraph, subject);
+        assert.equal(ns["dc:title"], "Title without references");
```

Comment:
Did you know that if you write this,

```suggestion
        assert(ns["dc:title"] === "Title without references");
```

and the assertion fails, power-assert will tease apart the expression and show you all intermediate values, just like in pytest assertions?

There is nothing wrong with `assert.equal` of course, so you don't need to go change all your assertions, but it might still be nice to know that you can do the pytest thing in the frontend, too.

Reply:
Great, I didn't know that!